src/apps/schedule/service/schedule.py

```diff
@@ -490,9 +500,85 @@ async def get_events_by_user(
                 PublicEventByUser(
                     applet_id=applet_id,
                     events=[
-                        PublicEvent(**event.dict()) for event in all_events
+                        self._convert_to_dto(event=event)
+                        for event in all_events
                     ],
                 )
             )
 
         return events
+
+    def _convert_to_dto(self, event: EventFull) -> ScheduleEventDto:
+        """Convert event to dto."""
+        return ScheduleEventDto(
+            id=event.id,
+            entityId=event.activity_id if event.activity_id else event.flow_id,
+            timers=TimerDto(
+                timer=HourMinute(
+                    hours=event.timer.seconds // 3600,
+                    minutes=event.timer.seconds // 60 % 60,
+                )
+                if event.timer_type == TimerType.timer
+                else None,
+                idleTimer=HourMinute(
+                    hours=event.timer.seconds // 3600,
+                    minutes=event.timer.seconds // 60 % 60,
+                )
+                if event.timer_type == TimerType.idle
+                else None,
+            ),
+            availabilityType=AvailabilityType.AlwaysAvailable
+            if event.periodicity.type == PeriodicityType.always
+            else AvailabilityType.ScheduledAccess,
+            availability=EventAvailabilityDto(
+                oneTimeCompletion=event.one_time_completion,
+                periodicityType=event.periodicity.type,
+                timeFrom=HourMinute(
+                    hours=event.start_time.hour,
+                    minutes=event.start_time.minute,
+                ),
+                timeTo=HourMinute(
+                    hours=event.end_time.hour, minutes=event.end_time.minute
+                ),
+                allowAccessBeforeFromTime=event.access_before_schedule,
+                startDate=event.periodicity.start_date,
+                endDate=event.periodicity.end_date,
+            ),
+        )
+
+    async def get_events_by_user_and_applet(
+        self, user_id: uuid.UUID, applet_id: uuid.UUID
+    ) -> PublicEventByUser:
+        """Get all events for user in applet."""
+
+        applet = await AppletsCRUD().get_applet_by_roles(
+            user_id=user_id,
+            applet_id=applet_id,
+            roles=[
+                Role.RESPONDENT,
+            ],
+        )
+
+        if not applet:
+            raise NotFoundError(
+                message=f"User {user_id} "
+                f"does not have access to applet {applet_id}"
+            )
+
+        user_events: list = await EventCRUD().get_all_by_applet_and_user(
+            applet_id=applet_id,
+            user_id=user_id,
+        )
+
+        general_events: list = await EventCRUD().get_general_events_by_user(
+            applet_id=applet_id, user_id=user_id
+        )
```

Comment:
I would avoid using regular `list` type hints because they define the list as a list of undefined items.
I recommend remove it completely or define the item type inside the list